Subject: [PATCH] 0008-dinput
---
Index: dlls/dinput/dinput_private.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/dlls/dinput/dinput_private.h b/dlls/dinput/dinput_private.h
--- a/dlls/dinput/dinput_private.h	(revision 675959b3a58df18c21fb918e7961f509827a035e)
+++ b/dlls/dinput/dinput_private.h	(date 1717507289278)
@@ -57,6 +57,9 @@
 extern HRESULT hid_joystick_enum_device( DWORD type, DWORD flags, DIDEVICEINSTANCEW *instance, DWORD version, int index );
 extern HRESULT hid_joystick_create_device( struct dinput *dinput, const GUID *guid, IDirectInputDevice8W **out );
 
+extern HRESULT gamepad_enum_device( DWORD type, DWORD flags, DIDEVICEINSTANCEW *instance, DWORD version );
+extern HRESULT gamepad_create_device( struct dinput *dinput, const GUID *guid, IDirectInputDevice8W **out );
+
 struct DevicePlayer {
     GUID instance_guid;
     WCHAR username[MAX_PATH];
Index: dlls/dinput/dinput.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/dlls/dinput/dinput.c b/dlls/dinput/dinput.c
--- a/dlls/dinput/dinput.c	(revision 675959b3a58df18c21fb918e7961f509827a035e)
+++ b/dlls/dinput/dinput.c	(date 1717510312142)
@@ -282,6 +282,7 @@
 
     if (IsEqualGUID( &GUID_SysKeyboard, guid )) hr = keyboard_create_device( impl, guid, &device );
     else if (IsEqualGUID( &GUID_SysMouse, guid )) hr = mouse_create_device( impl, guid, &device );
+    else if (IsEqualGUID( &GUID_Joystick, guid )) hr = gamepad_create_device( impl, guid, &device );
     else hr = hid_joystick_create_device( impl, guid, &device );
 
     if (FAILED(hr)) return hr;
@@ -372,6 +373,9 @@
 
     if (device_class == DI8DEVCLASS_ALL || device_class == DI8DEVCLASS_GAMECTRL)
     {
+        hr = gamepad_enum_device( type, flags, &instance, impl->dwVersion );
+        if (hr == DI_OK && try_enum_device( device_type, callback, &instance, context, flags ) == DIENUM_STOP)
+            return DI_OK;
         do
         {
             hr = hid_joystick_enum_device( type, flags, &instance, impl->dwVersion, i++ );
